### PR TITLE
Added missing expectations for EnginMasterTests

### PR DESCRIPTION
**Pull request by daschuer**
This fixes warnings reported in https://github.com/mixxxdj/mixxx/issues/11308

### CODE REVIEW

File: src/test/enginemastertest.cpp [hunk 1]
```diff
@@ -73,6 +73,10 @@ TEST_F(EngineMasterTest, SingleChannelOutputWorks) {
     EXPECT_CALL(*pChannel, isPflEnabled())
             .Times(1)
             .WillOnce(Return(false));
+    EXPECT_CALL(*pChannel, collectFeatures(_))
+            .Times(1);
+    EXPECT_CALL(*pChannel, postProcess(160000))
+            .Times(1);
```
Comment (Swiftb0y) [inline src/test/enginemastertest.cpp:79]: Do we really expect these to be called or is that just an artifact of the internals? 
If so, there are better ways of dealing with that. 
https://github.com/google/googletest/blob/3d787f5a0d58cfc37a0563bb15647a0d8aa2c1bf/docs/gmock_cook_book.md#knowing-when-to-expect-useoncall

Comment (daschuer) [reply src/test/enginemastertest.cpp:79]: It is just when the call is missing, it is likely a regression. I stumbled over it when digging into #11257

Comment (daschuer) [reply src/test/enginemastertest.cpp:79]: I don't really understand why ONE_CALL is an alternative. 

> So use ON_CALL by default, and only use EXPECT_CALL when you actually intend to verify that the call is made

In this case I want the later, so I think the solution is OK.

Comment (Swiftb0y) [reply src/test/enginemastertest.cpp:79]: Mhmm, the thing is that the other calls actually make sense. I don't understand why we need to verify that `collectFeatures` and `postProcess` calls have been made or what they even do or why its important for the test that they have been made. Perhaps you can shed some light on that?

Comment (daschuer) [reply src/test/enginemastertest.cpp:79]: For my understanding the purpose of these test is to detect significant changes in the code that may introduce regressions. All output should make the same calls. 
The tests are pointless in terms of "testing a contract" or such which is the original purpose of a Mock. 

I am currently working on finding the bug reported in https://github.com/mixxxdj/mixxx/issues/11257 where any additional test condition is welcome. But actually the main issue was the disturbing warning. This is fixed.

**Review by Swiftb0y — APPROVED** [review]
I guess its questionable how needed these are, but as long as they don't make it worse, I don't see a reason to reject this.

**Comment by daschuer** [comment]
@Swiftb0y So this can be merged?

**Comment by Swiftb0y** [comment]
Yes, I just wanted to give somebody else the opportunity to comment before merging.

**Comment by daschuer** [comment]
Thank you.

**Pull request merged**